chunk.c

```diff
@@ -2917,10 +2917,10 @@ void	_dmalloc_chunk_log_stats(void)
 		  BLOCK_SIZE, ALLOCATION_ALIGNMENT);
   
   /* general heap information with blocks */
-  dmalloc_message("heap address range: %p to %p, %ld bytes",
+  dmalloc_message("heap address range: %p to %p, %llu bytes",
 		  _dmalloc_heap_low, _dmalloc_heap_high,
-		  (unsigned long)_dmalloc_heap_high -
-		  (unsigned long)_dmalloc_heap_low);
+		  (unsigned long long)(unsigned PNT_ARITH_TYPE)_dmalloc_heap_high -
```

Comment:
unsigned long long was a no-no back in the day because some 32 bit systems needed to use the high bit of the lower word (sic) meaning that they were 63 bit numbers.  I kid you not.  So I'd rather this just be a cast to PNT_ARITH_TYPE.

Reply:
Wow, I didn't know that, thanks!

The problem is that it's hard to find a `printf` size modifier for `PNT_ARITH_TYPE`. If we want to be portable to the widest range of possible platforms (so probably ANSI C, and for most of them `sizeof(void*)==sizeof(long)` is probably true), `PNT_ARITH_TYPE` can't be larger than `long`, so printing it with `%lx`/`%ld` is okay. On the other hand, if we want this to work on 64-bit Windows, `long` is already too small, so even if we determine the correct (non-portable) type using a `./configure` test, finding out how to print it is still a problem. I think that ANSI C has appropriately-sized `ptrdiff_t` and `size_t`, but no portable way of printing them, either.

There is probably no beautiful solution for this. I can work on a `configure` test for the format string where pointer-sized integers have to be printed without `%p`, but code readability may suffer.